pkg/manifestreader/manifestloader.go

```diff
@@ -6,13 +6,18 @@ package manifestreader
 import (
 	"io"
 
+	"github.com/spf13/cobra"
 	"k8s.io/kubectl/pkg/cmd/util"
+	"sigs.k8s.io/cli-utils/cmd/flagutils"
+	"sigs.k8s.io/cli-utils/pkg/common"
+	"sigs.k8s.io/cli-utils/pkg/inventory"
 )
 
 // ManifestLoader is an interface for reading
 // and parsing the resources
 type ManifestLoader interface {
 	ManifestReader(reader io.Reader, path string) (ManifestReader, error)
+	GetInvInfo(cmd *cobra.Command, args []string) (inventory.Info, error)
```

Comment:
We shouldn't mix the new Loader interface with the ManifestLoader interface. As mentioned in another comment, lets keep the new interface and the implementation with the command.